src/actions.js

```diff
@@ -127,11 +127,15 @@ export const cut = (state: Types.StoreState<*>) => ({
   cut: true
 });
 
-export function paste<Cell: Types.CellBase>(state: Types.StoreState<Cell>) {
-  if (PointSet.isEmpty(state.copied)) {
-    return null;
-  }
-  const minPoint = PointSet.min(state.copied);
+export async function paste<Cell: Types.CellBase>(
+  state: Types.StoreState<Cell>
+) {
+  const text = await readTextFromClipboard();
+  if (!text) return null;
+  const matrix = Matrix.parse(text);
+  const copied = PointMap.fromMatrix<any>(matrix);
```

Comment:
No need to provide `any` here.
If Flow insists on a type argument it should be `Cell`.

Reply:
Using `Cell` causes a bunch of other errors. `any` was the best I could get, unfortunately.